### ISSUE TITLE
No sound on Samsung soundbar

### ISSUE THREAD

**Issue by mjf2708** [owntone/owntone-server#1465]
I am using Owntone 28.3, and I have a music library set up under OMV 5.6.26-1 (Usul) on an RPi 3B+ running Buster. The soundbar is a Samsung HW-S61A, which supports Airplay 2.

There is no sound from the soundbar using the Owntone / OMV library, even though the progress bars in both iOS Remote and SmartThings show that music is playing.

I have disabled IPv6, as well as raop for this device.

I get sound OK through a different Airplay 2 speaker (Apple Homepod Mini) using my Owntone / OMV library. I also get sound OK through the soundbar from Apple Music on macOS (Monterey 12.3.1).

**Comment by ejurgensen** [comment]
Thanks for adding the issue. If other people observe the same on other Airplay devices then please add a note here.

**Comment by thundershare** [comment]
yep confirm no sound from Samsung Hw-Q990B(soundbar) through Airplay 2. Same device works correctly with chromecast.
OWNTONE - VERSION 28.4 on RPi 3B+ Buster

**Comment by ejurgensen** [comment]
Strange that this only seems to appear with Samsung. I imagine they don't make their Airplay 2 implementation themselves, but get it from Apple.

**Comment by asaf2298** [comment]
Actually I have exactly the same problem with a b&w formation wedge speaker and the latest version of Owntone. This system include Sonos roam, Jbl link 200 and the Wedge. The Sonos and the Jbl play fine, but the wedge shown as playing airplay in his app while being silent.

**Comment by ejurgensen** [comment]
Thanks for adding a note about it. Can you confirm that this is an issue even with ipv6 disabled in the config?

It would be great to investigate and fix this, but I don't know anyone with these devices, and they are a bit too pricey to buy without really needing them.

**Comment by asaf2298** [comment]
Thank you and sure.
I don't mind trying everything you ask for, but I won't be near that system for some time.
I will update the moment I can confirm.

**Comment by thundershare** [comment]
> Thanks for adding a note about it. Can you confirm that this is an issue even with ipv6 disabled in the config?
> ....
i made a try with ip6 enable or disable but nothing change.. strange things: no error in the log.
If you need some specific test i can execute it.
OWNTONE - VERSION 28.5 on Debian

**Comment by ejurgensen** [comment]
Thanks for checking. If the device doesn't return any errors to OwnTone then you wouldn't see anything in the log. It is pretty complicated to debug this stuff, so I would need to have a test device myself.

**Comment by thundershare** [comment]
@ejurgensen 1 more info: seems that owntone(icon on web interface is on) has connected but is not true on soundbar. And volume up and down not work, so seems not be a only streaming problem.

**Comment by wniess** [comment]
Same issue with Samsung HW-Q995B soundbar and OwnTone 28.5 (on Debian Linux PC).

Tried disabling IP6 as well as raop - didn't help.

OwnTone seems to be streaming (progress bar is moving). When changing the volume in OwnTone, the soundbar shows the changed volume level on its display. When connecting the Samsung SmartThings app with the soundbar, the title of the currently played song is shown (--> the soundbar receives some data from OwnTone). All the time the soundbar is silent, nothing can be heard at all.

Streaming from an iPhone to the soundbar works well. Also streaming from OwnTone to a Shairport device. But: Streaming from iTunes (12.12.7.1 on Windows 11) behaves the same as OwnTone, also streaming seems to proceed, but no sound from the soundbar.

Any help is very appreciated! Thank you very much in advance!!

**Comment by vmonkey** [comment]
I experience the same issue, HW-S61B soundbar. Is there anything I can do to find the roof cause of the problem?

**Comment by ejurgensen** [comment]
Not really, it's a bit of reverse engineering task

**Comment by dodsonw** [comment]
FWIW, I am also experiencing the same issue with a HW-Q800A Soundbar.  Owntone's logs show no connection issues no matter what settings I mess with (IP6, RAOP, etc).  All other functionality works as expected - I just don't have any audio output.  I can even see in the SmartThings app that the Soundbar is "playing" the track from an airplay source.

Spent quite a bit of time looking over the network traffic and I can confirm that nothing looks abnormal but I am definitely no expert when it comes to the Airplay protocol...

The only weird thing I noticed is that there is an RTP stream present when playing from Owntone, but when I (successfully) airplay from another Apple device, I don't see any RTP stream.


Server - v28.8 Docker Container on Debian 11 Host

**Comment by helgek** [comment]
Same issue with Samsung Ultra Slim Soundbar HW-S801B. I also tried the Home Assistant Apple TV integration and raised a bug report there as well. Maybe this is an explanation why it doesn't work: https://community.roonlabs.com/t/implementation-of-airplay-2/229509/5 . Since the Samsung Soundbars are probably not backwards compatible to Airplay 1 and there might not be any working third party implementation to stream AirPlay2 audio to AirPlay2 receivers anytime soon this issue will possibly remain unsolved for now.

**Comment by ejurgensen** [comment]
> there might not be any working third party implementation to stream AirPlay2 audio

OwnTone does stream AirPlay 2 so that is not the reason.

**Comment by ejurgensen** [comment]
I'm just saying the reason is something else. Since it hasn't been found yet I am not so optimistic that it will be.

**Comment by dr-leo** [comment]
Same with JBL Charge 5 wi-fi. Device appears in the Apple Remote app. But when I click on play, it disappears and no sound is ever played. - I haven't deactivated ipv6 nore played with raop.

Here is the log:


[2025-03-17 21:37:33] [  LOG] [     worker (-1)]     dacp: Player returned an error for start after pause
[2025-03-17 21:37:47] [  LOG] [     worker (-1)]     daap: Error: Request to remove non-existent or ad-hoc session. BUG!
[2025-03-17 21:38:26] [  LOG] [     player (-1)]     cast: Device 'JBL Charge 5 Wi-Fi' could not launch app id '85CDB22F' nor '0F5096E8' - aborting
[2025-03-17 21:38:26] [  LOG] [     player (-1)]   player: The Chromecast device 'JBL Charge 5 Wi-Fi' failed to activate
[2025-03-17 21:38:39] [  LOG] [     player (-1)]     cast: Device 'JBL Charge 5 Wi-Fi' could not launch app id '85CDB22F' nor '0F5096E8' - aborting
[2025-03-17 21:38:39] [  LOG] [     player (-1)]   player: The Chromecast device 'JBL Charge 5 Wi-Fi' failed to activate
[2025-03-17 21:38:39] [  LOG] [     worker (-1)]     dacp: Speakers de/activation failed!
[2025-03-17 21:44:53] [  LOG] [    owntone (-1)]     mdns: Avahi Resolver failure: service 'JBL Charge 5 Wi-Fi' type '_airplay._tcp' proto 0: Timeout reached
[2025-03-17 21:44:53] [  LOG] [    owntone (-1)]     mdns: Avahi Resolver failure: service '94B3F7B1F20D@JBL Charge 5 Wi-Fi' type '_raop._tcp' proto 0: Timeout reached

**Comment by dmac001** [comment]
Similar but related issue with Samsung Q990F.  The soundbar is shown as an output device.  When you select and play a song, Owntone returns;  

Request failed (status: 400 Bad Request http://192.168.0.5:3689/api/outputs/44010667484485)

The log shows,

[2025-11-19 15:15:24] [  LOG] [   player (1824)]     raop: ANNOUNCE request failed in session startup: 403 Forbidden
[2025-11-19 15:15:24] [  LOG] [   player (1824)]   player: The AirPlay 1 device 'Q-Series Soundbar' failed to activate
[2025-11-19 15:15:24] [  LOG] [   worker (1818)]     dacp: Speakers de/activation failed!
[202


I've verified this behavior through the web UI and my iPhone.  

Note, from the iPhone Music app, I can Airplay stream directly to the Samsung Q990F soundbar.  This may indicate something with Owntown?

**Comment by ejurgensen** [comment]
Yes, it is clearly something in OwnTone, but as you can see from the above, it has not been possible to determine what the cause is

**Comment by dmac001** [comment]
Just hoping there was a new solution.  I can try to help troubleshoot this issue, running 29.0 on Ubuntu.

**Comment by ejurgensen** [comment]
Sorry, don't think there is one around the corner. It would be pretty complex to troubleshoot. Basically you have to mitm the Music app and the speaker, compare against OwnTone and by trial and error or by injecting changes find the difference that's the cause.

**Comment by DerPicknicker** [comment]
Hi @ejurgensen , 

I captured the Packages on my Samsung Soundbar (from Mac to Samsung Speaker + Music Assistant to Samsung Soundbar). I can also capture the packages from own tone. Would you mind if you can support us to build that: https://github.com/music-assistant/cliairplay? 

I captured via Wireshark with a working example, and we see the encrypted Data. What is missing currently for @bradkeifer (who is mainly working on this) the encryption of the Data. Is Owntone sending encrypted Data and decrypting incoming Data? 

Highly appreciating your Response!

**Comment by ejurgensen** [comment]
Yes, Airplay 2 is encrypted, so capturing packets is not so useful. You can write to me at espenjurgensen@gmail.com if you want some pointers on what you could do. 

> Would you mind if you can support us to build that

Not sure what you mean?

**Comment by DerPicknicker** [comment]
Hey @ejurgensen , 

thank you for your quick response. @bradkeifer is building that library: https://github.com/music-assistant/cliairplay which is used in MusicAssistant. I guess it will be easier if we made a small mailing list or something like that. Do you have Discord? We already have some discussions on discord so maybe you want to join? Email is also absolutely fine...

**Comment by ejurgensen** [comment]
Yes I am ejurgensen on Discord.

**Comment by ejurgensen** [comment]
If any of you Samsung-owners are still around and up for it, it would be great if you could help test OwnTone with PTP support, which is available in this [branch](https://github.com/owntone/owntone-server/tree/airplay_ptp1). It seems to have fixed a similar issue with Sonos Era ref issue #1853, so I'm hoping it also fixes Samsung.

Remember to make sure OwnTone uses Airplay 2, either by building with `./configure --enable-preferairplay2` or setting `raop_disable=true` in the logfile for the speaker.

**Comment by helgek** [comment]
@ejurgensen I have great news for you: It works 🎉🎉🎉🎉🎉🎉 Thank you so much 🙌🙌🙌 

Tested with Samsung Soundbar S801B and I had to use ./configure --enable-preferairplay2

I also played successfully on multiple speakers in parallel, the Samung Soundbar + 2 Sonos (older Sonos One  and IKEA Symfonisk picture frame), for all speakers it showed "AirPlay2" when hovering over the AirPlay activate icon of every speaker at http://localhost:3689/#/outputs . The two Sonos speakers used to work before on AirPlay1 protocol as well, but they also support AirPlay2 and play the music.

**side note**: On the output select page http://localhost:3689/#/outputs the status change from active->inactive or inactive->active of a speaker after clicking on the AirPlay icon doesn't automatically update (from black->greyed out or vice versa). After every activation/deactivation I have to reload the browser page to see the actual status in Owntone.

Thank you again!!! I'm looking forward to soon also have this fix in my other HomeAssistant integrations (Music Assistant, Apple TV, I just discovered that Owntown also shows up in Home Assistant as an integration). This is a major improvement for so many users!

**Comment by bradkeifer** [comment]
Thanks so much for this work @ejurgensen   
I expect to get sometime later this week to work on getting this latest OwnTone work incorporated into Music Assistant along with some other fixes.

**Comment by DerPicknicker** [comment]
Hey @ejurgensen , 

I just build on macOS and it works but I can confirm that the build is failing with chrome cast enabled. I tested on the latest Samsung Soundbar (HW-Q995F) which is working fine. Also my Musicframes from Samsung (HW-LS60D) are working fine.. 

@bradkeifer .. Let me know when those fixes are arriving in MusicAsisstant so we can check if the delays created on MA-Side can cause some problems.

**Comment by dmac001** [comment]
Greetings,

Reinstalling/upgrading to the latest owntone git release (not PTP branch),
owntone now works with my Samsung Q990F soundbar.   The only difference is
I added --enable-preferairplay2 to .configure and rebuilt.  I suspect I
needed --enable-preferairplay2 all along.  As such, I won't be trying the
PTP version.  Thank you for your ongoing support,

Don

**Comment by helgek** [comment]
In case that matters: I also didn't use a specific PTP release to build owntone. I just used the latest owntone-server
